tools/CMakeLists.txt

```diff
@@ -57,14 +57,22 @@ add_executable(${RELLIC_XREF}
   "xref/Xref.cpp"
 )
 
+include(FetchContent)
+FetchContent_Declare(cpp-httplib
+  GIT_REPOSITORY https://github.com/yhirose/cpp-httplib.git
+  GIT_TAG        v0.20.0
+)
+FetchContent_MakeAvailable(cpp-httplib)
+set(CPP_HTTPLIB_INCLUDE_DIRS "${cpp-httplib_SOURCE_DIR}")
+
 find_path(CPP_HTTPLIB_INCLUDE_DIRS "httplib.h")
 target_include_directories(${RELLIC_XREF} PRIVATE ${CPP_HTTPLIB_INCLUDE_DIRS})
 
 target_link_libraries(${RELLIC_XREF}
   PRIVATE
     "${PROJECT_NAME}_cxx_settings"
     "${PROJECT_NAME}"
-    gflags::gflags
+    gflags
 )
```

Comment:
IMO this should be gated behind an option like
```
option(RELLIC_USE_EXTERNAL_DEPENDENCIES "Use find_package for dependencies" OFF)
```
in `cmake/options.cmake`. At least let the user choose to provide their own cpp-httplib, even if it's untested by default. Most users won't care, which is fine, and it'll do the right thing because it's `OFF` by default.

Also, we should use the target `httplib::httplib` instead of `CPP_HTTPLIB_INCLUDE_DIRS`.